cli/internal/cmd/buffer.go

```diff
@@ -545,6 +586,346 @@ func newBufferImportCommand() *cobra.Command {
 	return cmd
 }
 
+func newBufferDeleteCommand() *cobra.Command {
+	deleteAll := false
+	force := false
+	tags := make(map[string]string)
+	excludeTags := make(map[string]string)
+
+	cmd := &cobra.Command{
+		Use:                   "delete ID ... | --tag KEY=VALUE ... | --all",
+		Short:                 "Delete buffers",
+		Long:                  `Delete buffers (soft-delete)`,
+		DisableFlagsInUseLine: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			validateExactlyOneMode := func(modes ...bool) bool {
+				count := 0
+				for _, mode := range modes {
+					if mode {
+						count++
+					}
+				}
+				return count == 1
+			}
+
+			if !validateExactlyOneMode(len(args) > 0, len(tags) > 0, deleteAll) {
+				log.Fatal().Msg("One of ID, --tag, or --all must be specified")
+			}
+
+			if len(args) > 0 {
+				var deleted []model.Buffer
+				for _, id := range args {
+					buffer := model.Buffer{}
+					relativeUri := fmt.Sprintf("v1/buffers/%s", id)
+					resp, err := controlplane.InvokeRequest(cmd.Context(), http.MethodDelete, relativeUri, nil, &buffer)
+
+					if resp.StatusCode == http.StatusNotFound {
+						return fmt.Errorf("buffer with ID %s not found", id)
+					} else if resp.StatusCode == http.StatusPreconditionFailed {
+						return fmt.Errorf("buffer with ID %s is already deleted", id)
+					}
+
+					if err != nil {
+						return err
+					}
+
+					deleted = append(deleted, buffer)
+				}
+
+				var formatted []byte
+				var err error
+				if len(deleted) == 1 {
+					formatted, err = json.MarshalIndent(deleted[0], "", "  ")
+				} else {
+					formatted, err = json.MarshalIndent(deleted, "", "  ")
+				}
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(formatted))
+
+			} else {
+				options := url.Values{}
+				for name, value := range tags {
+					options.Add(fmt.Sprintf("tag[%s]", name), value)
+				}
+				for name, value := range excludeTags {
+					options.Add(fmt.Sprintf("excludeTag[%s]", name), value)
+				}
+
+				performDeletion := force
+
+				if !force {
+					options.Add("softDeleted", strconv.FormatBool(false))
+					countUri := fmt.Sprintf("v1/buffers/count?%s", options.Encode())
+					count := 0
+					if _, err := controlplane.InvokeRequest(cmd.Context(), http.MethodGet, countUri, nil, &count); err != nil {
+						return err
+					}
+					if count > 0 {
+						log.Warn().Msgf("Deleting %d buffers. Use --force to delete without confirmation.", count)
```

Comment:
I think for an interactive operation, where the user is about to be prompted, I would not use `log`, since the message is prefixed with a timestamp. Maybe just roll the message into the prompt?